src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -140,6 +154,117 @@ bool Bank::is_remote(const LazyTensor& tensor) const {
     return false;
 }
 
+void Bank::serialize(std::ostream& stream) const {
+    using namespace ov::npuw::s11n;
+
+    LOG_INFO("Serializing weights bank...");
+    LOG_BLOCK();
+
+    std::lock_guard<std::mutex> guard(m_mutex);
+
+    // For now only a singular NPU or CPU device is supported
+    // Sanity check
+    NPUW_ASSERT(m_device_banks.size() == 1 && "Bank containing several devices can't be serialized");
+    auto it = m_device_banks.find("NPU");
+    auto it_cpu = m_device_banks.find("CPU");
+    NPUW_ASSERT((it != m_device_banks.end() || it_cpu != m_device_banks.end()) &&
+                "Only a singular NPU or CPU bank can be serialized");
+
+    if (it_cpu != m_device_banks.end()) {
+        it = it_cpu;
+    }
+
+    const auto& device_bank = it->second;
+    std::lock_guard<std::mutex> dev_guard(device_bank.mutex);
+    write(stream, it->first);
+    write(stream, device_bank.storage.size());
+    for (const auto& t_pair : device_bank.storage) {
+        write(stream, t_pair.second);
+    }
+
+    LOG_INFO("DONE.");
+}
+
+std::shared_ptr<Bank> Bank::deserialize(std::istream& stream, const std::shared_ptr<const ov::ICore>& core) {
+    using namespace ov::npuw::s11n;
+
+    LOG_INFO("Deserializing weights bank...");
+    LOG_BLOCK();
+
+    // For now only a singular NPU or CPU device is supported
+    std::string device;
+    read(stream, device);
+    NPUW_ASSERT((device == "NPU" || device == "CPU") && "Only a singular NPU or CPU bank can be serialized");
+    // Bank is assumed to be shared - thus no need for a unique name.
+    // FIXME: is that right? What about multi-model pipeline or several pipelines?
+    auto bank = ov::npuw::weights::bank("shared_serialized", core, device);
+    std::size_t bank_size = 0;
+    read(stream, bank_size);
+
+    for (std::size_t i = 0; i < bank_size; ++i) {
+        std::pair<ov::Tensor, std::size_t> p;
+        read(stream, p);
+        bank->add_element(p.second, p.first, device);
+    }
+
+    LOG_INFO("DONE.");
+
+    return bank;
+}
+
+void Bank::add_element(std::size_t uid, const ov::Tensor& tensor, const std::string& device) {
+    // This method is supposed to be used only during deserialization
+    // For now only a singular NPU or CPU device is supported
+    NPUW_ASSERT(device == "NPU" || device == "CPU");
+    std::lock_guard<std::mutex> guard(m_mutex);
+
+    auto& device_bank = m_device_banks[device];
+    std::lock_guard<std::mutex> dev_guard(device_bank.mutex);
+
+    device_bank.deserialized_storage[uid] = {ov::Tensor(tensor.get_element_type(), tensor.get_shape()), false};
+    tensor.copy_to(device_bank.deserialized_storage[uid].first);  // preserve data ownership
+}
+
+ov::Tensor Bank::get(std::size_t uid, const std::string& device) {
```

Comment:
As discussed, this signature could replace the old `get(LazyTensor)` but it needs to require `registreLT` first and not allocate anything